test/test_event_router.rb

```diff
@@ -229,6 +233,89 @@ def event_router
       end
     end
 
+    sub_test_case 'optimized filter' do
+      setup do
+        @record = { 'k' => 'v' }
+        @now = Engine.now
+      end
+
+      test 'call optimized filter when filter plugin that has #filter' do
+        event_router.add_rule('test', filter)
+
+        assert_rr do
+          mock(filter).filter('test', @now, @record) { @record }
+          event_router.emit('test', @now, @record)
+        end
+      end
+
+      test 'call optimized filter when filter plugin that has #filter_with_time' do
```

Comment:
"when the filter plugin implements `#filter_with_time` without `#filter_stream`"